dockstore-common/src/test/java/io/dockstore/common/yaml/DockstoreYamlTest.java

```diff
@@ -198,6 +198,32 @@ void testMissingPrimaryDescriptor() {
         }
     }
 
+    @Test
+    void testRelativePrimaryDescriptor() {
+        try {
+            final String content = DOCKSTORE12_YAML.replaceAll("primaryDescriptorPath: /", "primaryDescriptorPath: ");
+            DockstoreYamlHelper.readDockstoreYaml(content, true);
+            fail("Invalid dockstore.yml not caught");
+        } catch (DockstoreYamlHelper.DockstoreYamlException e) {
+            // check that the error message contains the name of the property and an appropriate adjective
+            assertTrue(e.getMessage().contains("primaryDescriptor"));
+            assertTrue(e.getMessage().contains("the path must be an absolute path to be valid"));
+        }
+    }
+
+    @Test
+    void testRelativeTestDescriptorPaths() {
```

Comment:
Just curious, did you check the DB for how often relative test parameter paths are present? In the Slack thread, I think you only posted relative primary descriptor instances.

Just wondering if this might end up breaking more workflows than we thought?

Reply:
In the database, there are currently 13 published workflows with relative test parameter paths. The first `sum` column is the number of versions and the second sum is the number of relative test parameter paths.

```
      organization       |        repository         |         workflowname         | sum | sum 
-------------------------+---------------------------+------------------------------+-----+-----
 aofarrel                | goleft-wdl                |                              |   1 |   1
 aofarrel                | upon-thine-inputs         | upon-thine-inputs            |   1 |   1
 dockstore-testing       | wes-testing               | single-descriptor-with-input |   1 |   1
 hloucks                 | BWA_WDL                   |                              |   1 |   1
 hloucks                 | FCS-WDL                   |                              |   1 |   1
 large-scale-gxe-methods | gem-workflow              | gem-workflow                 |   6 |   6
 large-scale-gxe-methods | magee-workflow            | magee-workflow               |   4 |   4
 large-scale-gxe-methods | metagem-workflow          | metagem-workflow             |   1 |   1
 large-scale-gxe-methods | regem-workflow            | regem-workflow               |   3 |   3
 phylyc                  | gatk4-somatic-snvs-indels | MultiSampleMutect2           |   2 |   2
 phylyc                  | gatk4-somatic-snvs-indels | Mutect2                      |   2 |   2
 shwetagopaul92          | GatherWGS                 |                              |   1 |   1
 shwetagopaul92          | ScatterWGS                |                              |   4 |   5
```